### ISSUE TITLE
Division by Zero error

### ISSUE THREAD

**Issue by zawilliams** [nystudio107/craft-imageoptimize-sharp#6]
The same error from this issue: https://github.com/nystudio107/craft-imageoptimize-imgix/issues/3 is needs the same fix from here: https://github.com/nystudio107/craft-imageoptimize-imgix/commit/4d14fe9d36cc7add443c56cd11857ee0f350a125 

Just ran into the same bug but with this package.

**Comment by zawilliams** [comment]
@khalwat I made a PR to address this issue here: #7.

**Comment by zawilliams** [comment]
@khalwat any chance this can get pushed through and added to a release soon? Currently having to patch it for a client.

**Comment by khalwat** [comment]
Yep will get it out shortly!

**Comment by zawilliams** [comment]
Thanks @khalwat!

**Comment by khalwat** [comment]
Released as:

Craft 3: Version 1.0.11 -> https://github.com/nystudio107/craft-imageoptimize-sharp/releases/tag/1.0.11

Craft 4: Version 4.0.1 -> https://github.com/nystudio107/craft-imageoptimize-sharp/releases/tag/4.0.1

**Comment by zawilliams** [comment]
Thanks!

**Comment by zawilliams** [comment]
Also, I assume these will get updated and a new release will happen with [nystudio107/craft-imageoptimize](https://github.com/nystudio107/craft-imageoptimize) as well?

**Comment by khalwat** [comment]
So a new release of the plugin itself isn't needed. You can just do:

```
composer update nystudio107/craft-imageoptimize-sharp
```

These were designed to be separate packages that could be updated independently of the plugin itself.

But yes, sometime soon a new version of the plugin will come out as well.

**Comment by zawilliams** [comment]
Got it - will do. Thanks @khalwat!

**Comment by alexmillman23** [comment]
Issue seems to remain. Version 4.0.10

   "name": "Exception",
    "message": "Division by zero",
    "code": 0,
    "error": "Division by zero",
    "exception": "DivisionByZeroError",
    "file": "/app/vendor/nystudio107/craft-imageoptimize/src/services/Optimize.php",
    "line": 649,

Had to go and manually delete pdf and webp files from our asset buckets. otherwise, the whole volume is inaccessible.

**Comment by khalwat** [comment]
@alexmillman23 did you do:

```
composer update nystudio107/craft-imageoptimize-sharp
```

Also, are you using Sharp? Or are you using something else?

**Comment by khalwat** [comment]
@alexmillman23 you are using an old version of ImageOptimize. Please update to the latest version, that fix you're looking for is implemented in version 4.0.11 -- and the latest version is 4.0.14
